### DISCUSSION TITLE
how to fetch multiple markdown file to create a single page with nextjs

### DISCUSSION THREAD

**Discussion by aj-ya** [markdoc/markdoc#135] [Q&A] [answered]
I'm creating a docs website for a UI library, and I want to fetch multiple markdown files together from another folder (component/usage, props, etc) such that I can tab switch to navigate between them.
Currently, I'm doing this with partials, but the catch with partials is that I cant use the frontmatter data from the partials.
```
{% pagetabs %}
    {% tabpane type="usage" %}
        {% partial file="button/usage.md" /%}
    {% /tabpane %}
    {% tabpane type="props" %}
        {% partial file="button/props.md" /%}
    {% /tabpane %}
{% /pagetabs %}
```
Is there any better way to achieve this functionality?

**Comment by mfix-stripe** [comment, marked as answer]
@aj-ya partials can have their own frontmatter, and you can also pass the partials their own `variables={}`. Likewise, you can edit the content directly in the `pagetabs` or `tabpane` `transform` functions. 

I like the pattern you have started above though!

**Reply by aj-ya** [reply]
thanks for the answer

**Comment by naderkhalil** [comment]
Hello! Im stuck trying to do the same thing, @mfix-stripe could you elaborate your answer?

Specifically, if I have a few posts with frontmatter, I want to make a "library" page (which wouldn't be markdown), displaying the frontmatter attributes of each post. How could I go about querying the content of all the md files to access their frontmatter?

Something like `pageProps.markdoc?.frontmatter.pageTitle` but for multiple pages. 

Thank you!!

**Reply by nkohari-stripe** [reply]
You'd need to do this at build time -- specifically, you would look at the frontmatter of every document in your corpus, grab the `pageTitle` property, and write them all somewhere that can be accessed at runtime.

There's a (pretty hacky) example of how to do this on [my website](https://github.com/nkohari/nate.io/blob/master/build/templates/catalog.ts). It uses a custom Vite plugin to gather metadata for all the documents and writes them to a JS module, which can then be imported in the app.
